src/hooks/useUserProfile.tsx

```diff
@@ -1,11 +1,35 @@
-import React, { createContext, PropsWithChildren, useEffect, useState } from 'react';
+import React, { createContext, PropsWithChildren, useCallback, useEffect, useState } from 'react';
 import { createUserProfile, getUserProfile } from '../api';
 import { UserProfile } from '../types';
 
-export const UserProfileContext = createContext<UserProfile | null>(null);
+export const UserProfileContext = createContext<{
+  userProfile: UserProfile | null;
+  refreshUserProfile: (() => Promise<UserProfile | null | undefined>) | null;
+  loading: boolean;
+}>({ userProfile: null, refreshUserProfile: null, loading: false });
 
 export function UserProfileProvider({ children, address }: PropsWithChildren<{ address: string | null | undefined }>) {
   const [userProfile, setUserProfile] = useState<UserProfile | null>(null);
+
+  const [loading, setLoading] = useState(false);
+
+  const refreshUserProfile = useCallback(async () => {
+    if (loading) {
+      return userProfile;
+    }
```

Comment:
This one is a bit unnecessary. I think it's better to handle this situation in the component that is using the callback